Compiler/FunctionCompiler.py

```diff
@@ -171,15 +171,16 @@ def get_token_after_array_access(self, offset=0):
 
     def compile_array_assignment(self, token_id):
         # int id[a][b][c]... = {1, 2, 3, ...};
+        # or int id[a][b][c]... = "\1\2\3...";
         # or int id[a][b][c]... = {{1, 2}, {3, 4}, ...};
         # or array assignment: id = {1, 2, 3, ...};
         self.parser.check_current_tokens_are([Token.ASSIGN])
         if self.parser.current_token().data != "=":
             raise BFSyntaxError("Unexpected %s when assigning array. Expected ASSIGN (=)" % self.parser.current_token())
 
-        assert self.parser.current_token().type == Token.ASSIGN and self.parser.current_token().data == "="
-        self.parser.check_current_tokens_are([Token.ASSIGN, Token.LBRACE])
-        self.parser.advance_token(1)  # skip to LBRACE
+        assert self.parser.next_token().type in [Token.LBRACE, Token.STRING]
```

Comment:
Same problem here, need to use check_current_tokens_are to inform the user where the error is